ops/testing.py

```diff
@@ -458,6 +463,30 @@ def add_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
         self._charm.on[relation_name].relation_joined.emit(
             relation, remote_unit.app, remote_unit)
 
+    def update_unit_status(self, relation_id: int, remote_unit_name: str,
+                           remote_unit_status: str) -> None:
```

Comment:
I think the unit status is one of the fields that we intend to deprecate from goal-state. Is there a particular use case that you need it for? Otherwise I would avoid adding it to OF.